src/utils.rs

```diff
@@ -70,3 +70,13 @@ pub fn teardown_user_io<B: Backend>(terminal: &mut Terminal<B>) -> color_eyre::R
     terminal.clear()?;
     Ok(())
 }
+
+#[inline]
+/// Simply calls `which` to check if a binary exists
+pub fn binary_exists(binary: &str) -> bool {
+    std::process::Command::new("which")
```

Comment:
I think `which` isn't built-in, even though it will probably be present in most Linux systems. If I am not mistaken, we should opt to use `command -v` or `type` (preferably `command -v`), as they are built-in and will work in the same manner, i.e., will return a non zero return code if the binary isn't resolvable.

Reply:
My shell (nushell) has neither of `command -v` nor `type`

Reply:
You are right! After further research, non-POSIX shells like Nushell and Fish doesn't necessarily have `which`, so using `command -v` isn't the best solution. 

However, this being a hard dependency and the program `which` being an external dependency in most shells, how about using the crate [which](https://crates.io/crates/which)? 

You know I prefer not to add crates indiscriminately, but as we need to use a dependency either way, I think a Rust crate trumps assuming user's setups.

From what I saw, importing the crate, and using

```
which("b4").is_ok()
```

will do the trick.

Reply:
No problem, and also i've just changed the implementation to use the `which` crate